homeassistant/components/rest/binary_sensor.py

```diff
@@ -96,18 +96,18 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     if username and password:
         if config.get(CONF_AUTHENTICATION) == HTTP_DIGEST_AUTHENTICATION:
-            auth = HTTPDigestAuth(username, password)
+            auth = httpx.DigestAuth(username, password)
         else:
-            auth = HTTPBasicAuth(username, password)
+            auth = (username, password)
     else:
         auth = None
 
-    rest = RestData(method, resource, auth, headers, payload, verify_ssl, timeout)
-    rest.update()
+    rest = RestData(hass, method, resource, auth, headers, payload, verify_ssl, timeout)
```

Comment:
It would be good to eventually refactor and move the `RestData` class out of the sensor platform. We don't want platforms to depend on each other.

Reply:
I didn't do it because the PR was already large. I'll do that in a followup

Reply:
https://github.com/home-assistant/core/pull/41980